model/riscv_insts_vext_mem.sail

```diff
@@ -0,0 +1,1490 @@
+
+  foreach (i from 0 to (num_elem - 1)) {
+    if mask[i] then { /* active elements */
+      vstart = to_bits(16, i);
+      let elem_offset = i * load_width_bytes;
+      match ext_data_get_addr(rs1, to_bits(sizeof(xlen), elem_offset), Write(Data), width_type) {
+        Ext_DataAddr_Error(e)  => { ext_handle_data_check_error(e); return RETIRE_FAIL },
+        Ext_DataAddr_OK(vaddr) =>
+          if check_misaligned(vaddr, width_type)
+          then { handle_mem_exception(vaddr, E_SAMO_Addr_Align()); return RETIRE_FAIL }
+          else match translateAddr(vaddr, Write(Data)) {
+            TR_Failure(e, _)     => { handle_mem_exception(vaddr, e); return RETIRE_FAIL },
+            TR_Address(paddr, _) => {
+              let eares : MemoryOpResult(unit) = mem_write_ea(paddr, load_width_bytes, false, false, false);
+              match (eares) {
+                MemException(e) => { handle_mem_exception(vaddr, e); return RETIRE_FAIL },
+                MemValue(_) => {
+                  let res : MemoryOpResult(bool) = mem_write_value(paddr, load_width_bytes, vs3_val[i], false, false, false);
+                  match (res) {
+                    MemValue(true)  => (),
+                    MemValue(false) => internal_error(__FILE__, __LINE__, "store got false from mem_write_value"),
+                    MemException(e) => { handle_mem_exception(vaddr, e); return RETIRE_FAIL }
+                  }
+                }
+              }
+            }
+          }
+      }
+    }
+  };
+
+  vstart = EXTZ(0b0);
+  RETIRE_SUCCESS
+}
+
+function clause execute(VSETYPE(vm, rs1, width, vs3)) = {
+  let load_width_bytes = vlewidth_bytesnumber(width);
+  let EEW = load_width_bytes * 8;
+  let EEW_pow = vlewidth_pow(width);
+  let SEW_pow = get_sew_pow();
+  let LMUL_pow = get_lmul_pow();
+  let EMUL_pow = EEW_pow - SEW_pow + LMUL_pow;
+  let num_elem = get_num_elem(EMUL_pow, EEW);
+
+  if illegal_vd_unmasked() then { handle_illegal(); return RETIRE_FAIL };
+
+  process_vse(vm, vs3, load_width_bytes, rs1, EMUL_pow, num_elem)
+}
+
+mapping vsetype_mnemonic : vlewidth <-> string = {
+  VLE8        <-> "vse8.v",
+  VLE16       <-> "vse16.v",
+  VLE32       <-> "vse32.v",
+  VLE64       <-> "vse64.v"
+}
+
+mapping clause assembly = VSETYPE(vm, rs1, width, vs3)
+  <-> vsetype_mnemonic(width) ^ spc() ^ vreg_name(vs3) ^ sep() ^ "(" ^ reg_name(rs1) ^ ")" ^ maybe_vmask(vm)
+
+/* ************************** Vector Load Strided Normal (nf=0, mop=10) ************************** */
+union clause ast = VLSETYPE : (bits(1), regidx, regidx, vlewidth, regidx)
+
+mapping clause encdec = VLSETYPE(vm, rs2, rs1, width, vd) if haveRVV()
+  <-> 0b000 @ 0b0 @ 0b10 @ vm @ rs2 @ rs1 @ encdec_vlewidth(width) @ vd @ 0b0000111 if haveRVV()
+
+val process_vlse : forall 'b 'n 'p, ('b in {1, 2, 4, 8}) & ('n >= 0). (bits(1), regidx, int('b), regidx, regidx, int('p), int('n)) -> Retired effect {escape, rmem, rmemt, rreg, undef, wmv, wmvt, wreg}
```

Comment:
We merged the segment and non-segment ones. For strided and non-strided ones, because they are two main types of load/store instructions and have separate encoding, maybe keeping them separately can make it clearer to read and understand the spec.